src/F2_autoreport.py

```diff
@@ -645,6 +645,35 @@ def create_app(results_json):
     # Rename the timestamp column from 'index' to 'timestamp'
     df_all_demands = df_all_demands.rename(columns={"index": "timestamp"})
 
+    # Collect the keys of various resources (PV, Wind, etc.)
+    resources = results_json[ENERGY_PRODUCTION]
+
+    # List of resources (includes DSOs, which need to be removed)
+    res_keys = list(resources.keys())
+    for res in res_keys:
+        if 'DSO_' in res:
+            del resources[res]
+
+    # List of resources (with the DSOs deleted)
+    res_keys = list(resources.keys())
+
+    # The below loop will add all the resources to the dict_for_plots dictionary, including the timeseries values
+    for resource in res_keys:
+        dict_for_plots["supplies"].update(
+            {resource: results_json[ENERGY_PRODUCTION][resource]["timeseries"]}
```

Comment:
```suggestion
            {resource: results_json[ENERGY_PRODUCTION][resource][TIMESERIES]}
```
Whenever you type a string as a dict key, go look into `constant_json_string.py` if there is not variable for that string. If not, you can create a variable for your string and look for occurrences in the *.py files and replace them by the variable